cmd/templ/main.go

```diff
@@ -75,6 +75,7 @@ examples:
 func generateCmd(args []string) {
 	cmd := flag.NewFlagSet("generate", flag.ExitOnError)
 	fileNameFlag := cmd.String("f", "", "Optionally generates code for a single file, e.g. -f header.templ")
+	outputFlag := cmd.String("o", "", "Optionally set an output for the Go code. If -f is set this should be a filename if not this should be a directory.")
```

Comment:
```
If -f is set this should be a filename, otherwise, it should be a directory.
```

I don't see this being validated or checked anywhere. It seems like I could run `go generate -f test.templ -o test.templ` and overwrite my templ file, or generate something that doesn't match the expected `_templ.go` pattern.